mrmustard/lab_dev/wires.py

```diff
@@ -200,22 +208,50 @@ def __add__(self, other: Wires) -> Wires:
         for m in all_modes:
             self_row = self.id_array[self._modes.index(m)] if m in self.modes else np.zeros(4)
             other_row = other.id_array[other._modes.index(m)] if m in other.modes else np.zeros(4)
-            # if np.any(np.where(self_row > 0) == np.where(other_row > 0)):
-            #     raise ValueError(f"wires overlap on mode {m}")
+            if np.any(np.where(self_row > 0) == np.where(other_row > 0)):
+                raise ValueError(f"wires overlap on mode {m}")
             modes_rows[m] = [s if s > 0 else o for s, o in zip(self_row, other_row)]
         combined_array = np.array([modes_rows[m] for m in sorted(modes_rows)])
-        return self._from_data(None, sorted(modes_rows), np.ones_like(combined_array))
         return self._from_data(combined_array, sorted(modes_rows), np.ones_like(combined_array))
 
     def __bool__(self) -> bool:
-        return True if self.ids else False
+        return True if len(self.ids) > 0 else False
 
     def __getitem__(self, modes: Iterable[int] | int) -> Wires:
         "A view of this Wires object with wires only on the given modes."
         modes = [modes] if isinstance(modes, int) else modes
         idxs = tuple(list(self._modes).index(m) for m in set(self._modes).difference(modes))
         return self._view(masked_rows=idxs)
 
+    def _mode(self, mode: int) -> np.ndarray:
+        "A slice of the id_array matrix at the given mode."
+        return np.maximum(0, self.id_array[[self._modes.index(mode)]])[0]
+
+    def __lshift__(self, other: Wires) -> Wires:
+        return (other.dual >> self.dual).dual  # how cool is this
+
+    def __rshift__(self, other: Wires) -> Wires:
+        r"""Returns a new Wires object with the wires of self and other combined as two
+        components in a circuit: the output of self connects to the input of other wherever
+        they match.
+        """
+        all_modes = set(self.modes) | set(other.modes)
+        new_id_array = np.zeros((len(all_modes), 4), dtype=np.int64)
+        for i, m in enumerate(sorted(all_modes)):
```

Comment:
```suggestion
        for i, m in enumerate(all_modes):
```
`all_modes` is already sorted since it's a set

Reply:
<img width="242" alt="Screenshot 2024-01-26 at 10 55 07 AM" src="https://github.com/XanaduAI/MrMustard/assets/8944955/7bf0ea17-6330-4024-9300-a45cf1ff7da5">
it's not for me

Reply:
it's random:
<img width="307" alt="Screenshot 2024-01-26 at 10 55 59 AM" src="https://github.com/XanaduAI/MrMustard/assets/8944955/1770a825-cd78-488f-ac40-8cb9e09d6f5a">

Reply:
mmmh okay. I was thrown off by what you get if you do `set([8, 4, 0])`, which prints `{0, 4, 8}`. But you are right, I checked online and they are unsorted. Probably only the repr is

Reply:
And I think it's a feature of sets, not a downside.